walker.go

```diff
@@ -0,0 +1,84 @@
+package ftp
+
+import (
+	"fmt"
+	"strings"
+)
+
+//Walker traverses the directory tree of a remote FTP server
+type Walker struct {
+	serverConn *ServerConn
+	root       string
+	cur        item
+	stack      []item
+	descend    bool
+}
+
+type item struct {
+	path  string
+	entry Entry
+	err   error
+}
+
+// Next advances the Walker to the next file or directory,
+// which will then be available through the Path, Stat, and Err methods.
+// It returns false when the walk stops at the end of the tree.
+func (w *Walker) Next() bool {
```

Comment:
Why not return the item ?

Reply:
Was aiming for parity with the Go standard library; their Walker.Step() function returns a bool and I wanted to keep the usage of this implementation as close to that as possible. 

It also means that the caller can only use the Walker object to manage the traversing of the directory tree and the encapsulation of the item within the Walker object maintains a clear separation between the caller and the package.